core/app/models/spree/tax/application.rb

```diff
@@ -0,0 +1,51 @@
+module Spree
+  module Tax
+    class Application
+      def initialize(order, output)
+        @order = order
+        @output = output
+      end
+
+      def perform!
+        %i(line_items shipments).each do |items|
+          order.send(items).each do |item|
+            rates = output.send(items).select { |i| i.id == item.id }
+
+            tax_adjustments = item.adjustments.select(&:tax?)
+
+            active_adjustments = rates.map do |rate|
+              update_adjustment(item, rate)
+            end
+
+            # Remove any tax adjustments tied to rates which no longer match.
+            unmatched_adjustments = tax_adjustments - active_adjustments
+            item.adjustments.destroy(unmatched_adjustments)
+          end
+        end
+      end
+
+      private
+
+      attr_reader :order, :output
+
+      def update_adjustment(item, output_item)
+        tax_adjustment = item.adjustments.detect do |adjustment|
+          adjustment.source == output_item.tax_rate
+        end
+
+        tax_adjustment ||= item.adjustments.new(
+          source: output_item.tax_rate,
```

Comment:
Ah! This is why `Spree::TaxRate` still plays a role. 

Adjustments need a `source` so they can be individually re-calculated. However, with any order-level, API-based taxation solution, this will never happen. 

We might need to think about something like a `Spree::Tax` model, similar to adjustments but NOT adjustments.